src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidator.cs

```diff
@@ -71,5 +89,29 @@ public IEnumerable<ModelValidationResult> Validate(ModelValidationContext valida
 
             return Enumerable.Empty<ModelValidationResult>();
         }
+
+        private static IDictionary<Type, ValidationAttributeAdapterFactory> AttributeAdapters =
+            ValidationAttributeAdapterTable.AttributeFactories;
+
+        private string GetErrorMessage(ModelMetadata metadata, IModelMetadataProvider metadataProvider)
+        {
+            if (Attribute == null)
+            {
+                throw new ArgumentNullException(nameof(Attribute));
+            }
+
+            var attrType = Attribute.GetType();
+            ValidationAttributeAdapterFactory factory;
+
+            if (AttributeAdapters.TryGetValue(attrType, out factory))
+            {
+                var t = factory(Attribute, _stringLocalizer);
+                return t.GetErrorMessage(metadata, _modelMetadataProvider);
+            }
+            else
+            {
+                throw new NotImplementedException($"Error message localizer not defined for {attrType.Name}");
```

Comment:
Should not exist. No `Exception` should occur if we encounter an unrecognized attribute.